### ISSUE TITLE
Items: right click functions are not run if left click is held down (but it works if it's the opposite)

### ISSUE THREAD

**Issue by Zughy** [minetest/minetest#13581]
##### Minetest version
```
5.7.0, 5.7.0-dev
```

##### Summary
Actions on `on_secondary_use` and `on_place` are not run if you're holding down the left mouse button. However, you can run `on_use` even if you're holding down the right button.  

I expect actions to run no matter what keys I'm currently holding down. In my use case, I have a weapon that it's supposed to fire a charged attack when LMB is released and that uses the gunsight with RMB. However, if I'm charging the shot, I can't zoom in/out, which is frustrating.

##### Steps to reproduce
```lua
minetest.register_node("test:node", {
    description = "Test node",
    tiles = {"server_favorite.png"},
    inventory_image = "server_favorite.png",
    groups = {cracky = 3},
    
    on_use = function(itemstack, user, pointed_thing)
    	minetest.chat_send_all("On use")
    end,
    
    on_secondary_use = function(itemstack, user, pointed_thing)
    	minetest.chat_send_all("On secondary use")
    end,
    
    on_place = function(itemstack, user, pointed_thing)
    	minetest.chat_send_all("On place")
    end
})
```

click around

**Comment by sfan5** [comment]
I do not understand the description and can't find anything wrong with the test code.

Holding down dig or place will trigger just once. Holding down both should not have a defined behavior, as it does not make sense.

**Comment by Zughy** [comment]
> Holding down both should not have a defined behavior, as it does not make sense.

Some games might implement mechanics that require to hold down left and then click right. As in the example reported in my first message. It does make sense for some games

**Comment by sfan5** [comment]
That does not mean it is the right mechanic. Items are intended for interacting with objects and nodes, not as a replacement for an input system where you can assign defined actions to clicking or holding.
There isn't even any feedback for holding down place/dig (how are you handling that?).

**Comment by SmallJoker** [comment]
> Actions on `on_secondary_use` and `on_place` are not run if you're holding down the left mouse button. However, you can run `on_use` even if you're holding down the right button.

This is about inconsistent mouse event handling. I can confirm this.

* RMB (place) is disabled while LMB is held down (punch/use).
* LBM (punch/use) can be used as many times as desired while RMB (place) is held down.

I think this should be fixed.
EDIT: Relevant code: https://github.com/luanti-org/luanti/blob/5.14.0/src/client/game.cpp#L3238-L3257

---

I don't know how relevant this is, but the following topic also crossed my mind while thinking about your issue.
The two buttons also differ in how the events are fired:

* LMB (punch/use) is fired exactly once when pressing the key.
   * Meanwhile tools have a `full_punch_interval` and yet, `on_use` is only executed once.
* RMB (place) triggers a repeated `on_place` callback depending on the setting `repeat_place_time`.

**Comment by Zughy** [comment]
> The two buttons also differ in how the events are fired

That's true but I fear that changing this behaviour might break some content (I'm pretty sure it'd break my weapons mod, for instance)
